From 54e3aca84e571559915998aa6cc05e5ac37c043b Mon Sep 17 00:00:00 2001
From: Andy Shevchenko <andriy.shevchenko@linux.intel.com>
Date: Mon, 18 Mar 2019 21:47:09 +0300
Subject: [PATCH] ACPI / utils: Drop reference in test for device presence
Git-commit: 54e3aca84e571559915998aa6cc05e5ac37c043b
Patch-mainline: v5.1-rc2
References: bsc#1051510

When commit 8661423eea1a ("ACPI / utils: Add new acpi_dev_present
helper") introduced acpi_dev_present(), it missed the fact that
bus_find_device() took a reference on the device found by it and
the callers of acpi_dev_present() don't drop that reference.

Drop the reference on the device in acpi_dev_present().

Fixes: 8661423eea1a ("ACPI / utils: Add new acpi_dev_present helper")
Signed-off-by: Andy Shevchenko <andriy.shevchenko@linux.intel.com>
Reviewed-by: Hans de Goede <hdegoede@redhat.com>
Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Acked-by: Takashi Iwai <tiwai@suse.de>

---
 drivers/acpi/utils.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/acpi/utils.c
+++ b/drivers/acpi/utils.c
@@ -798,6 +798,7 @@ bool acpi_dev_present(const char *hid, c
 	dev = bus_find_device(&acpi_bus_type, NULL, &match,
 			      acpi_dev_present_cb);
 
+	put_device(dev);
 	return !!dev;
 }
 EXPORT_SYMBOL(acpi_dev_present);
